website/content/docs/release-notes/1.9.0.mdx

```diff
@@ -0,0 +1,121 @@
+
+### Customized username generation for database dynamic credentials
+
+This feature enables customization of username for database dynamic credentials. This feature helps customers better manage and correlate usernames for various actions such as troubleshooting, etc. Vault 1.9 supports Postgres, MSSQL, MySQL, Oracle, MongoDB.
+
+### Customizable HTTP headers for Vault
+
+This [features](https://github.com/hashicorp/vault/pull/12485) allows security operators to configure [custom headers](/docs/configuration/listener/tcp) to HTTP root path (/) and API endpoints (/v1/*), in addition to the previously supported UI paths through the server HCL configuration file. If desired, this will enable customers to add custom security headers at these paths, thereby maintaining an improved security posture.
+
+### Support for IBM s390X CPU architecture
+
+This feature adds support for Vault to run on the IBM s390x architecture via the [equivalent binary](https://releases.hashicorp.com/vault/1.9.0).
+
+### Namespace API lock
+
+This [feature](/docs/concepts/namespace-api-lock) allows namespace administrators to flexibly control operations such as locking APIs from child namespaces to which they have access. This enables them to restrict access to their domain in a multi-tenant environment and perform break-glass procedures in times of emergency to protect a cluster from within their child namespace.
+
+### Vault Terraform Provider v3
+
+We have upgraded the [Vault Terraform Provider](https://registry.terraform.io/providers/hashicorp/vault/latest/docs) to the latest version of the [Terraform Plugin SDKv2](https://www.terraform.io/docs/extend/sdkv2-intro.html) to leverage new features.
+
+### Azure Secrets Engine
+
+The following enhancement are included:
+
+* Added `use_microsoft_graph_api` configuration parameter is added to use with Microsoft Graph API. We are targeting to remove Azure Active Directory API by June 30, 2022.
+* Rotate root
+
+### Customized metadata for KV
+
+This [enhancement](https://github.com/hashicorp/vault/pull/12907) provides the ability to set version-agnostic custom key metadata for Vault KVv2 secrets via a metadata endpoint. This custom metadata will also be visible in the UI.
+
+## UI Enhancements
+### Expanding the UI for more DB secrets engines
+
+We have been adding support for DB secrets engines in the UI over the past few releases. In the Vault 1.9 release, we have added support for [Oracle](https://github.com/hashicorp/vault/pull/12752) and [ElasticSearch](https://github.com/hashicorp/vault/pull/12672) and [PostgresSQL](https://github.com/hashicorp/vault/pull/12945) database secrets engines in the UI.
+
+### PKI certificate metadata
+
+The [PKI Secrets Engine](/docs/secrets/pki) now displays additional PKI certificate metadata in the UI, such as date issued, date of expiry, serial number, and subject/name.
+
+## GA Support for Previously Supported Tech Preview Features
+
+### Key Management Secrets(KMS) engine - GCP
```

Comment:
GCP support for KMSE is a new feature in 1.9. It was not released as a tech preview in 1.8. I suggest moving this into the ADP Enhancements section.

Reply:
Done